code/modules/mapfluff/ruins/spaceruin_code/meatderelict.dm

```diff
@@ -0,0 +1,133 @@
+	default_raw_text = "Ive overclocked the power generators to add that needed juice to the experiment, though theyre a bit unstable."
+
+/obj/item/paper/crumpled/fluff/meatderelict/fridge
+	name = "engineer complaint"
+	default_raw_text = "Whoever keeps stealing my fucking icecream from my fridge, I swear I will actually fuck you up. It is not cheap to get this delicious icecream here, nor is it for you. <b>And dont touch my snacks in the drawer!</b>"
+
+/obj/machinery/computer/terminal/meatderelict
+	upperinfo = "COPYRIGHT 2500 NANOSOFT-TM - DO NOT REDISTRIBUTE - Now with audio!" //not that old
+	content = list(
+		"Experimental Test Satellite 37B<br/>Nanotrasen™️ approved deep space experimentation lab<br/><br/>Entry 1:<br/><br/>Subject - \[Species 501-C-12\]<br/>Date - \[REDACTED\]<br/>We have acquired a biological sample of unknown origins \[Species 501-C-12\] from an NT outpost on the far reaches. Initial experiments have determined the sample to be of an unknown creature never previously recorded. It weighs approximately 7 grams and seems to be docile. Initial examinations determine that it is an extremely fast replicating creature that can alter its physiology to take multiple differing shapes. \[Recording Terminated\]<br/>- Dr. Phil Cornelius",
+		"Entry 2:<br/><br/>Subject - \[Species 501-C-12\]<br/>Date - \[REDACTED\]<br/>The creature responds to electrical stimuli. It has failed to respond to Light, Heat, Cold, Oxygen, Plasma, CO2, Nitrogen. It, within moments, seemed to have generated muscle tissue within its otherwise shapeless form and moved away from the source of electricity. Feeding the creature has been a simple matter, it consumed just about any form of protein. It appears to rapidly digest and convert forms of protein into more of itself. Any undigestible products are simply left alone. Will continue to monitor creature and provide reports to Nanotrasen Central Command. \[Recording Terminated\]<br/>- Dr. Phil Cornelius",
+		"Entry 3:<br/><br/>Subject - \[Species 501-C-12\]<br/>Date - \[REDACTED\]<br/>Any attempts at contacting Nanotrasen has failed. I've never seen anything like it. I... I don't think I'm going to survive much longer, I can hear it pushing on my room door. If anyone reads this, let my family know that I- \[Loud crash\]<br/>GET BACK \[Gunshots\]<br/>AHHHHHHHHHHHH \[Recording Terminated\]<br/>- Dr. Phil Cornelius"
+	)
+
+/obj/machinery/door/puzzle/meatderelict
+	name = "lockdown door"
+	desc = "A beaten door, still sturdy. Impervious to conventional methods of destruction, must be a way to open it nearby."
+	base_icon_state = "danger"
+	icon_state = "danger_closed"
+	puzzle_id = "md_prevault"
+
+/mob/living/basic/meteor_heart/opens_puzzle_door
+	var/id
+
+/mob/living/basic/meteor_heart/opens_puzzle_door/death(gibbed)
+	. = ..()
+	addtimer(CALLBACK(src, PROC_REF(send_signal)), 2.5 SECONDS)
+
+/mob/living/basic/meteor_heart/opens_puzzle_door/proc/send_signal()
+	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_PUZZLE_COMPLETED, id)
```

Comment:
im not sure if i understand

Reply:
A rule for signals is that they should have only one source, but `COMSIG_GLOB_PUZZLE_COMPLETED` is sent from a whole load of different places.
You've got two options here, which are:
- Investigate a way to make your puzzles respond to a variety of signals as required.
- Make a global proc which fires this signal so that "technically" it is only sent from one place.

Ninja is essentially our headcoder authority in charge of _why_ this is and may or may not tell us that the second solution is shitcode which we also shouldn't do, but it _is_ the simpler option.

Reply:
so do i just do
```
/proc/puzzle_completed(id)
SEND_GLOBAL_SIGNAL(COMSIG_GLOB_PUZZLE_COMPLETED, id)
```

Reply:
Yeah that's basically what I mean